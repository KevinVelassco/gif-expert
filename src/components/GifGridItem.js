import React from 'react';
import PropTypes from 'prop-types';

export const GifGridItem = ({ title, url }) => {
    return (
        <div className="card  animate__animated animate__flip">
            <div className="card__imgContainer">
                <img src={url} alt={title} className="card__img" />
            </div>
        </div>
    )
}

GifGridItem.propTypes = {
    title: PropTypes.string.isRequired,
    url: PropTypes.string.isRequired

}
