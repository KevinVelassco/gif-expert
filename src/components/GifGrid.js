import React, { useState, useEffect } from 'react';
import { getGifs } from '../helpers/getGifs';
import { GifGridItem } from './GifGridItem';

export const GifGrid = ({ category }) => {

    const [images, setimages] = useState([]);

    useEffect(() => {
        getGifs(category)
            .then(setimages)
    }, [category]);

    return (
        <>
            <h2>{category}</h2>
            <div className="panel">
                {
                    images.map(img => (
                        <GifGridItem
                            key={img.id}
                            {...img} />
                    ))
                }
            </div>
        </>
    );
}