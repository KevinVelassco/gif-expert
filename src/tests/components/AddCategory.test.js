import React from 'react';
import { shallow } from "enzyme";
import { AddCategory } from "../../components/AddCategory";

describe('Test component <AddCategory />', () => {

    const setCategories = jest.fn();
    let wrapper;

    beforeEach(() => {
        jest.clearAllMocks();
        wrapper = shallow(<AddCategory setCategories={setCategories} />);
    });

    test('Debe mostrar el componente correctamente', () => {
        expect(wrapper).toMatchSnapshot();
    });

    test('Simular eventos change y submit', () => {
        const value = 'Hola Mundo';

        wrapper.find('input').simulate('change', { target: { value } });
        wrapper.find('form').simulate('submit', { preventDefault() { } });

        expect(setCategories).toHaveBeenCalled();
        expect(wrapper.find('input').prop('value')).toBe('');
    });
});