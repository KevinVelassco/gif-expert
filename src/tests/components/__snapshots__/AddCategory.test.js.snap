// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Test component <AddCategory /> Debe mostrar el componente correctamente 1`] = `
<form
  onSubmit={[Function]}
>
  <input
    className="input animate__animated animate__flash"
    onChange={[Function]}
    type="text"
    value=""
  />
</form>
`;

exports[`Test componet <AddCategory /> Debe mostrar el componente correctamente 1`] = `
<form
  onSubmit={[Function]}
>
  <input
    className="input animate__animated animate__flash"
    onChange={[Function]}
    type="text"
    value=""
  />
</form>
`;
